spring-cloud-stream-test-support-internal/src/main/java/org/springframework/cloud/stream/test/junit/kafka/KafkaTestSupport.java

```diff
@@ -70,8 +68,14 @@
 	private static boolean hasFailedAlready = false;
 
 	static {
-		// check if either the environment or Java property is set to use embedded tests
-		defaultEmbedded = "true".equals(System.getenv(SCS_KAFKA_TEST_EMBEDDED)) || "true".equals(System.getProperty(SCS_KAFKA_TEST_EMBEDDED));
+		if (System.getProperty(SCS_EXTERNAL_SERVERS_REQUIRED) != null &&
```

Comment:
If someone specifies `SCS_EXTERNAL_SERVERS_REQUIRED` to **true** explicitly, then the resource existence check will make sure to fail with the following error:
```
java.lang.AssertionError: KAFKA IS NOT AVAILABLE
	at org.junit.Assert.fail(Assert.java:88)
```

The above check is only needed to make sure `SCS_EXTERNAL_SERVERS_REQUIRED` is not set to `true` and proceed to use the `embedded` server for tests.